arangod/Aql/Query.cpp

```diff
@@ -93,6 +94,7 @@ Query::Query(std::shared_ptr<transaction::Context> const& ctx,
       _options(options),
       _queryOptions(_vocbase.server().getFeature<QueryRegistryFeature>()),
       _trx(nullptr),
+      _startTimeStamp(TRI_microtime()),
       _startTime(currentSteadyClockValue()),
```

Comment:
why is this there twice now?

Reply:
The `_startTimeStamp` is the time stamp when the query was started, as measured by the system clock. We use this only to report this timestamp in the QueryList for the list of currently running and slow queries, when we convert it into a timestring such as `2020-07-01T01:23:45Z`.

The `_startTime` is the time stamp when the query was started, as measured by the steady clock. We use this to calculate the query's run time in several places. We can't use the system clock here, as it is not guaranteed to jump back and forth. We had one case in a unit test where we compared two system clock values to calculate the query runtime, and it was negative :cry: 

So we use the steady clock for any runtime measurements and elapsed time since start, and the system clock just for reporting the absolute time.
The steady clock values alone don't have a meaning or even a unit, they can use be used to calculate time differences.

That's why we now have two. We could get away with just one timestamp, at the expense of having negative runtime values here and there (which will make assertions fail).

Reply:
We could obviously get of `_startTimeStamp` in Query.cpp and only create it in QueryList.cpp when we first insert the query into the list of running queries. That should simplify it. Do you think this is worth doing?

Reply:
Separating them would be less head-scratching for readers, yes. Could also potentially rename `_startTime` to `_timerStart` or something to help disambiguate that it's not actually a "time" so much as a stopwatch value.